packages/auth-construct/src/construct.ts

```diff
@@ -712,18 +756,41 @@ export class AmplifyAuth
         output.mfaTypes = JSON.stringify(mfaTypes);
       }
     }
-
-    if (this.oauthMappings[authProvidersList.amazon]) {
-      output.amazonClientId = this.oauthMappings[authProvidersList.amazon];
+    const oauthMappings = this.providerSetupResult.oauthMappings;
+    if (oauthMappings[authProvidersList.amazon]) {
+      output.amazonClientId = oauthMappings[authProvidersList.amazon];
     }
-    if (this.oauthMappings[authProvidersList.facebook]) {
-      output.facebookClientId = this.oauthMappings[authProvidersList.facebook];
+    if (oauthMappings[authProvidersList.facebook]) {
+      output.facebookClientId = oauthMappings[authProvidersList.facebook];
     }
-    if (this.oauthMappings[authProvidersList.google]) {
-      output.googleClientId = this.oauthMappings[authProvidersList.google];
+    if (oauthMappings[authProvidersList.google]) {
+      output.googleClientId = oauthMappings[authProvidersList.google];
     }
-    if (this.oauthMappings[authProvidersList.apple]) {
-      output.appleClientId = this.oauthMappings[authProvidersList.apple];
+    if (oauthMappings[authProvidersList.apple]) {
+      output.appleClientId = oauthMappings[authProvidersList.apple];
+    }
+
+    if (this.providerSetupResult.providersList.length > 0) {
+      output.socialProviders = JSON.stringify(
+        this.providerSetupResult.providersList
+      );
+      // if any providers were defined, we must expose the oauth settings to the output
+      if (this.oAuthSettings) {
+        output.oauthDomain = `${this.domainPrefix}.auth.${
+          Stack.of(this).region
+        }.amazoncognito.com`;
+        output.oauthScope = JSON.stringify(
+          this.oAuthSettings.scopes?.map((s) => s.scopeName) ?? []
+        );
+        output.oauthRedirectSignIn = this.oAuthSettings.callbackUrls
+          ? this.oAuthSettings.callbackUrls.join(',')
+          : '';
+        output.oauthRedirectSignOut = this.oAuthSettings.logoutUrls
+          ? this.oAuthSettings.logoutUrls.join(',')
+          : '';
+        output.webClientId = this.resources.userPoolClient.userPoolClientId;
+        output.oauthResponseType = 'code';
```

Comment:
note: 'code' seems to be the default for hosted ui (old cli), when would this not be the case?

Reply:
is this still an open question?

Reply:
not anymore; 'token' is considered legacy and less secure than 'code' so we don't have a need for 'token' at the moment